src/field_theory/krull_topology.lean

```diff
@@ -287,3 +287,222 @@ begin
+  natural restriction map `res : (L ≃ₐ[K] L) → (E →ₐ[K] L)`. Moreover, this restriction map gives
+  a pushforward of `f` to an ultrafilter on `E →ₐ[K] L`. If `E/K` is finite, then this pushforward
+  is generated by `alg_hom_of_finite_dimensional_of_ultrafilter h_findim f`.  -/
+lemma alg_hom_of_finite_dimensional_of_ultrafilter_spec {K L : Type*} [field K] [field L]
+  [algebra K L] {E : intermediate_field K L} (h_findim : finite_dimensional K E)
+  (f : ultrafilter (L ≃ₐ[K] L)) : (f.map (λ σ : L ≃ₐ[K] L, σ.to_alg_hom.comp
+  (intermediate_field.val E)) : filter (E →ₐ[K] L)) =
+  pure (alg_hom_of_finite_dimensional_of_ultrafilter h_findim f) :=
+classical.some_spec
+  (@ultrafilter.eq_principal_of_fintype (E →ₐ[K] L) _
+  (f.map (λ σ, σ.to_alg_hom.comp (intermediate_field.val E))))
+
+
+lemma alg_hom_of_finite_dimensional_of_ultrafilter_functor {K L : Type*} [field K] [field L]
+  [algebra K L] {E : intermediate_field K L} (hE : finite_dimensional K E)
+  (f : ultrafilter (L ≃ₐ[K] L)) {F : intermediate_field K L} (hF : finite_dimensional K F)
+  (hEF : E ≤ F) : alg_hom_of_finite_dimensional_of_ultrafilter hE f =
+  (alg_hom_of_finite_dimensional_of_ultrafilter hF f).comp (subalgebra.inclusion hEF) :=
+  begin
+    set p_E := (λ σ : L ≃ₐ[K] L, σ.to_alg_hom.comp (intermediate_field.val E)) with p_E_def,
+    set p_F := (λ σ : L ≃ₐ[K] L, σ.to_alg_hom.comp (intermediate_field.val F)) with p_F_def,
+    set σ_E := alg_hom_of_finite_dimensional_of_ultrafilter hE f with σ_E_def,
+    set σ_F := alg_hom_of_finite_dimensional_of_ultrafilter hF f with σ_F_def,
+    have hσ_E := alg_hom_of_finite_dimensional_of_ultrafilter_spec hE f,
+    rw [← p_E_def, ← σ_E_def] at hσ_E,
+    have hσ_F := alg_hom_of_finite_dimensional_of_ultrafilter_spec hF f,
+    rw [← p_F_def, ← σ_F_def] at hσ_F,
+    set res : (F →ₐ[K] L) → (E →ₐ[K] L) := (λ ϕ, ϕ.comp (subalgebra.inclusion hEF))
+      with res_def,
+    have h_pF_pE_res : res ∘ p_F = p_E := rfl,
+    have h_maps_commute : ((f.map p_F).map res : filter (E →ₐ[K] L)) = f.map p_E,
+    { rw [ultrafilter.map_map, h_pF_pE_res] },
+    have hEf := alg_hom_of_finite_dimensional_of_ultrafilter_spec hE f,
+    rw [← σ_E_def, ← p_E_def] at hEf,
+    have hFf := alg_hom_of_finite_dimensional_of_ultrafilter_spec hF f,
+    rw [← σ_F_def, ← p_F_def] at hFf,
+    have hFf' : (ultrafilter.map p_F f) = (pure σ_F : ultrafilter (F →ₐ[K] L)) :=
+    ultrafilter.coe_inj.mp hFf,
+    rw [hEf, hFf', ultrafilter.map_pure] at h_maps_commute,
+    have h := filter.pure_injective h_maps_commute,
+    rw res_def at h,
+    exact h.symm,
+  end
+
+/-- For each `x : L`, the intermediate field `K(x)` is finite dimensional over `K`, so
+  `alg_hom_of_finite_dimensional_of_ultrafilter` gives a map `ϕₓ : K(x) → L`. We define
+  `function_of_ultrafilter h_int f` to be the function taking `x` to `ϕₓ(x)` for all `x : L`. -/
+noncomputable def function_of_ultrafilter {K L : Type*} [field K] [field L] [algebra K L]
```

Comment:
`function_of_ultrafilter` has the same generality as `alg_hom_of_ultrafilter`, so there's no reason to ever use the unbundled version. Could you either inline its definition or name it something `alg_hom_of_ultrafilter_aux` to make it clear that it's an auxiliary definition. That further means you should write an API for `alg_hom_of_ultrafilter`, rather relying on the one from `function_of_ultrafilter`.